src/panels/BasePanel.ts

```diff
@@ -124,10 +135,21 @@ function getMessageContext<TContent extends ContentId>(
         subscribeToMessages: (handler) => {
             webview.onDidReceiveMessage(
                 (message: object) => {
-                    if (!isValidMessage<ToVsCodeMessage<TContent>>(message)) {
+                    if (!isValidMessage<ToVsCodeMsgDef<TContent>>(message)) {
                         throw new Error(`Invalid message to VsCode: ${JSON.stringify(message)}`);
                     }
 
+                    const getTelemetryData = telemetryDefinition[message.command];
+                    if (getTelemetryData !== false) {
```

Comment:
@peterbom I am confused here , you are checking for !== false and then inside you check for === true which is like conflicting. Can you elaborate more on this ?

Reply:
@peterbom I got it now, for a second it was confusing. But still adding some comments here would be helpful for the readers. :)

Reply:
I agree, that doesn't read well. I'll see if I can write it more clearly!

Just to explain quickly for now, though: the value of `getTelemetryData` is of type `boolean | `(args: MsgArgs) => { [key: string]: string }`

We only emit telemetry if it's `true` or a function is specified, in other words if `getTelemetryData !== false`. Once inside that condition, it must be either `true` or be a function.

Reply:
Extracted this logic into a separate function with separate `return` statements for each condition. I _think_ this makes it clearer.

Reply:
looks good to me. Thank you.